sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -376,7 +376,22 @@ _nib_dr_entry_t *_nib_drl_add(const ipv6_addr_t *router_addr, unsigned iface)
 void _nib_drl_remove(_nib_dr_entry_t *nib_dr)
 {
     if (nib_dr->next_hop != NULL) {
+        _evtimer_del(&nib_dr->rtr_timeout);
         nib_dr->next_hop->mode &= ~(_DRL);
+#if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_DC)
+/*  When removing a router from the Default
+    Router list, the node MUST update the Destination Cache in such a way
+    that all entries using the router perform next-hop determination
+    again rather than continue sending traffic to the (deleted) router.
+    @see [RFC4861](https://datatracker.ietf.org/doc/html/rfc4861#section-6.3.5)
+*/
```

Comment:
```suggestion
        /*  When removing a router from the Default
            Router list, the node MUST update the Destination Cache in such a way
            that all entries using the router perform next-hop determination
            again rather than continue sending traffic to the (deleted) router.
            @see [RFC4861](https://datatracker.ietf.org/doc/html/rfc4861#section-6.3.5)
        */
```

Reply:
(nice catch btw!)